### PR TITLE
Gitlab new project

### PR DESCRIPTION
**Pull request by jverce**
This PR is in support of issue #376

* Add event source logic for New Project

### CODE REVIEW

File: components/gitlab/new-project.js [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+      type: "$.interface.timer",
+      default: {
+        intervalSeconds: 15 * 60, // 15 minutes
+      },
+    },
+    http: "$.interface.http",
+    db: "$.service.db",
+  },
+  hooks: {
+    async activate() {
```
Comment (tjk) [inline components/gitlab/new-project.js:20]: Hm, I think I like this pattern of using the activate hook for this. We should backport it to older components, although in older components we typically do emit a recent page (eg. for test event reasons). We should tackle that issue differently though -- this is good.

File: components/gitlab/new-project.js [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+      const opts = {
+        pageSize: 1,
+      };
+      const projects = await this.collectProjects(opts);
```
Comment (tjk) [inline components/gitlab/new-project.js:24]: Am I misreading? We only need the latest project here but it looks like you'd collect all the user's projects but with 1 per_page? Would be nice to just make call for the first page.

**Review by tjk — CHANGES_REQUESTED** [review]
https://github.com/PipedreamHQ/pipedream/pull/467/files#r496457562

File: components/gitlab/new-project.js (hunk 1 above)
Comment (jverce) [reply components/gitlab/new-project.js:20]: +1
We should definitely strive for consistent building blocks across the project.

File: components/gitlab/new-project.js (hunk 2 above)
Comment (jverce) [reply components/gitlab/new-project.js:24]: I think you're right, good catch!! I'll change this to use what `getCommits` already does (limit the amount of queried items).

Comment (jverce) [reply components/gitlab/new-project.js:25]: Indeed you were right. Just pushed a fix.

**Review by tjk — APPROVED** [review]

**Pull request merged**